core/services/ocr2/plugins/functions/plugin.go

```diff
@@ -68,7 +71,36 @@ func NewFunctionsServices(sharedOracleArgs *libocr2.OCR2OracleArgs, conf *Functi
 	}
 	listenerLogger := conf.Lggr.Named("FunctionsListener")
 	bridgeAccessor := functions.NewBridgeAccessor(conf.BridgeORM, FunctionsBridgeName, MaxAdapterResponseBytes)
-	functionsListener := functions.NewFunctionsListener(oracleContract, conf.Job, bridgeAccessor, pluginORM, pluginConfig, conf.Chain.LogBroadcaster(), listenerLogger, conf.MailMon, conf.URLsMonEndpoint)
+
+	// Threshold plugin
+	sharedThresholdOracleArgs := libocr2.OCR2OracleArgs{
```

Comment:
Simply `sharedThresholdOracleArgs := *sharedOracleArgs` to copy all fields to a new object.

Reply:
I want to create an independent object, not a pointer to sharedOracleArgs.  We need 2 distinct objects because the `ReportingPluginFactory` needs to be different.

Reply:
Yes, that's what my suggestion would do.